test/com/facebook/buck/jvm/kotlin/KotlinBuckConfigTest.java

```diff
@@ -17,168 +17,190 @@
-            .build();
+    Path kotlinRuntime = testDataDirectory
+        .resolve("faux_kotlin_home")
+        .resolve("libexec")
+        .resolve("lib")
+        .resolve("kotlin-compiler.jar");
+
+    Logger.get(KotlinBuckConfigTest.class).warn(testDataDirectory.resolve("faux_kotlin_home").toAbsolutePath().toString());
+    BuckConfig buckConfig = FakeBuckConfig.builder()
+        .setSections(ImmutableMap.of(
+            "kotlin",
+            ImmutableMap.of("kotlin_home", testDataDirectory.resolve("faux_kotlin_home").toAbsolutePath().toString(),
+                "external", "false")))
+        .build();
 
     KotlinBuckConfig kotlinBuckConfig = new KotlinBuckConfig(buckConfig);
-    String command = kotlinBuckConfig.getKotlinCompiler().get().getCommandPrefix(null).get(0);
-    assertEquals(command, kotlinCompiler.toString());
+    Path compilerJar = kotlinBuckConfig.getPathToCompilerJar();
+    assertEquals(kotlinRuntime, compilerJar);
   }
 
   @Test
-  public void testFindsKotlinRuntimeLibraryInPath() throws IOException {
+  public void testFindsKotlinCompilerLibraryInPath() throws IOException {
     // Get faux kotlinc binary location in project
     Path kotlinPath = workspace.resolve("bin");
     Path kotlinCompiler = kotlinPath.resolve("kotlinc");
     MoreFiles.makeExecutable(kotlinCompiler);
 
-    BuckConfig buckConfig =
-        FakeBuckConfig.builder()
-            .setEnvironment(
-                ImmutableMap.of(
-                    "PATH", kotlinPath.toString() + pathSeparator + System.getenv("PATH")))
-            .build();
+    BuckConfig buckConfig = FakeBuckConfig.builder()
+      .setSections(
+          ImmutableMap.of("kotlin", ImmutableMap.of("external", "true")))
+      .setEnvironment(
+        ImmutableMap.of("PATH",
+            kotlinPath.toString() + pathSeparator +
+            System.getenv("PATH")))
+      .build();
 
     KotlinBuckConfig kotlinBuckConfig = new KotlinBuckConfig(buckConfig);
-    Path runtimeJar = kotlinBuckConfig.getPathToRuntimeJar().getRight();
-    Assert.assertThat(
-        runtimeJar.toString(),
-        Matchers.containsString(workspace.getPath(".").normalize().toString()));
+    Path compilerJar = kotlinBuckConfig.getPathToCompilerJar();
+    Assert.assertThat(compilerJar.toString(),
+                      Matchers.containsString(workspace.getPath(".").normalize().toString()));
   }
 
   @Test
-  public void testFindsKotlinRuntimeInConfigWithAbsolutePath()
-      throws HumanReadableException, IOException {
+  public void testFindsKotlinStdlibJarInConfigWithAbsolutePath()
+      throws HumanReadableException, InterruptedException, IOException {
 
-    Path kotlinRuntime = workspace.resolve("lib").resolve("kotlin-runtime.jar");
+    Path kotlinRuntime = testDataDirectory
+        .resolve("faux_kotlin_home")
+        .resolve("libexec")
+        .resolve("lib")
+        .resolve("kotlin-stdlib.jar");
```

Comment:
i think tests are failing because of kotlin-stdlib / kotlin-runtime naming confusion